workspace/src/main/java/org/entcore/workspace/controllers/WorkspaceController.java

```diff
@@ -926,71 +925,70 @@ public void handle(JsonObject res) {
 				if ("ok".equals(status) && jo != null) {
 					boolean createThumbnails = !StringUtils.isEmpty(thumbSize) && (jo.getJsonObject("thumbnails") == null ||
 							StringUtils.isEmpty(jo.getJsonObject("thumbnails").getString(thumbSize)));
-					JsonObject thumbnails = createThumbnails ? new JsonObject().put(thumbSize, "") : new JsonObject();
-					folderManager.createThumbnailIfNeeded(jo, thumbnails, asyncDefaultResponseHandler ->
-					{
-						JsonObject result;
-						boolean failed = asyncDefaultResponseHandler.failed();
-						if (failed == true)
-						{
-							log.error(asyncDefaultResponseHandler.cause());
-							result = jo;
+					if(createThumbnails) {
+						// We launch the creation of the thumbnail if it was required but we won't send.
+						// Instead, we will send the original file. It is an optimization to cut the dependency on the thumbnail creation
+						// service.
+						final JsonObject thumbnails = createThumbnails ? new JsonObject().put(thumbSize, "") : new JsonObject();
+						folderManager.createThumbnailIfNeeded(jo, thumbnails, asyncDefaultResponseHandler -> {
+							boolean failed = asyncDefaultResponseHandler.failed();
+							if (failed == true) {
+								log.error("An error occurred while creating file thumbnail " + fileId, asyncDefaultResponseHandler.cause());
+							}
+							else {
+								log.debug("Successfully created thumbnail for file " + fileId);
+							}
+						});
+					}
+					String file;
+					if (!createThumbnails && thumbSize != null && !thumbSize.trim().isEmpty()) {
+						file = DocumentHelper.getThumbnails(jo).getString(thumbSize,
+								jo.getString("file"));
+						if(file == null || file.trim().isEmpty() == true) {
```

Comment:
```suggestion
						if(StringUtils.isEmpty(file)) {
```

Reply:
Du coup, autant simplifier le bloc de code complet :
```suggestion
file = (createThumbnails || StringUtils.isEmpty(thumbSize)) ? jo.getString("file") : DocumentHelper.getThumbnails(jo).getString(thumbSize,jo.getString("file"));

```
Par contre, j'émets de sérieuse réserve sur le fait de renvoyer l'image d'origine quand il existe un thumbnail d'une autre taille. Appelé en boucle depuis une liste côté front, ça fait du volume sur le réseau...
De plus, le mobile peut aussi être impacté.
Il vaudrait mieux renvoyer un autre thumbnail, si possible.